apps/builder/app/builder/features/settings-panel/use-settings-logic.ts

```diff
@@ -0,0 +1,44 @@
+import type { Instance } from "@webstudio-is/project-build";
+import store from "immerhin";
+import { type KeyboardEventHandler, useRef } from "react";
+import { instancesStore } from "~/shared/nano-states";
+
+type SettingUpdate = { setting: "label"; value: string };
+
+export const useSettingsLogic = ({
+  selectedInstance,
+}: {
+  selectedInstance: Instance;
+}) => {
+  const changes = useRef<Map<SettingUpdate["setting"], SettingUpdate["value"]>>(
+    new Map()
+  );
+
+  const set = ({ setting, value }: SettingUpdate) => {
+    switch (setting) {
+      case "label": {
+        changes.current.set(setting, value);
+      }
+    }
+  };
+
+  const update = () => {
+    store.createTransaction([instancesStore], (instances) => {
+      const instance = instances.get(selectedInstance.id);
+      if (instance !== undefined) {
+        instance.label = changes.current.get("label");
+      }
+    });
+  };
```

Comment:
updateLabel

Reply:
this function is meant for all the things, but its also internal and not exported/returned, and we don't know what we are going to need in the future, maybe we will indeed need separate functions for each update